app/lib/translation_service/deepl.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+class TranslationService::DeepL < TranslationService
+  include JsonLdHelper
+
+  def initialize(api_key)
+    super()
+
+    @api_key = api_key
+  end
+
+  def translate(text, source_language, target_language)
+    request(text, source_language, target_language).perform do |res|
+      case res.code
+      when 429
+        raise TooManyRequestsError
+      when 456
+        raise QuotaExceededError
+      when 200...300
+        transform_response(res.body_with_limit)
+      else
+        raise UnexpectedResponseError
+      end
+    end
+  end
+
+  private
+
+  def request(text, source_language, target_language)
+    req = Request.new(:post, 'https://api-free.deepl.com/v2/translate', form: { text: text, source_lang: source_language.upcase, target_lang: target_language, tag_handling: 'html' })
```

Comment:
Can we use an environment variable to specify the URL for DeepL API Pro?

Reply:
FYI `api.deepl.com`

Reply:
Done!